src/crypto/SecretKey.h

```diff
@@ -121,10 +121,15 @@ template <> struct KeyFunctions<PublicKey>
 
     static std::string getKeyTypeName();
     static bool getKeyVersionIsSupported(strKey::StrKeyVersionByte keyVersion);
+    static bool
+    getKeyVersionIsVariableLength(strKey::StrKeyVersionByte keyVersion);
     static PublicKeyType toKeyType(strKey::StrKeyVersionByte keyVersion);
     static strKey::StrKeyVersionByte toKeyVersion(PublicKeyType keyType);
-    static uint256& getKeyValue(PublicKey& key);
-    static uint256 const& getKeyValue(PublicKey const& key);
+    static uint256& getEd25519Value(PublicKey& key);
```

Comment:
This works in the context of this PR, but we should open an issue to clean up the `SignerKey` vs `PublicKey` abstractions:
the `convert` function is quite error prone, and the worst one is `verify(DecoratedSignature const& sig, SignerKey const& signerKey,
       Hash const& hash)` that will just crash if used with something that cannot be converted into a `PublicKey`.